presto-main/src/main/java/com/facebook/presto/operator/unnest/Unnester.java

```diff
@@ -122,4 +115,20 @@ protected final UnnestBlockBuilder getBlockBuilder(int index)
      * Get the length of the nested element at position {@code index}
      */
     protected abstract int getElementsLength(int index);
+
+    protected abstract boolean needToInsertNulls(int offset, int length, int maxTotalEntriesForBatch, int unnesterCount);
+
+    protected UnnestBlockBuilder getUnnestBlockBuilder(int index)
+    {
+        verify(index >= 0 && index < unnestBlockBuilders.length, format("index should be equal to or greater than 0 and less than %d, but it is %d", unnestBlockBuilders.length, index));
+
+        return unnestBlockBuilders[index];
+    }
+
+    protected void setUnnestBlockBuilder(UnnestBlockBuilder unnestBlockBuilder, int index)
```

Comment:
This method is used only in the constructors. It can be replaced with a constructor argument.

1. change constructor for Unnester to take a list of UnnestBlockBuilders
```
    protected Unnester(UnnestBlockBuilder... unnestBlockBuilders)
    {
        this.unnestBlockBuilders = requireNonNull(unnestBlockBuilders, "unnestBlockBuilders is null");
    }
```

2. Change constructor for ArrayUnnester

```
super(new GenericUnnestBlockBuilder(elementType));
```

3. Change constructor for MapUnnester

```
super(new GenericUnnestBlockBuilder(keyType), new GenericUnnestBlockBuilder(valueType));
```

4. Change constructor for ArrayOfRowsUnnester

```
    public ArrayOfRowsUnnester(Type... types)
    {
        super(createUnnestBlockBuilder(types));
        this.fieldCount = types.length;
    }

    private static UnnestBlockBuilder[] createUnnestBlockBuilder(Type[] types)
    {
        UnnestBlockBuilder[] builders = new UnnestBlockBuilder[types.length];

        for (int i = 0; i < types.length; i++) {
            builders[i] = new ArrayOfRowUnnestBlockBuilder(types[i]);
        }

        return builders;
    }
```